### PR TITLE
Support Gemma3/4 + Eagle3

### PR DESCRIPTION
**Pull request by pyc96**
## Motivation

This PR supports Gemma3/4 model with Eagel3 and fixes multiple bugs in current eagle3 implementation.

- Support aux layers embedding captures for both models. Fixes an issue when trying to capture the last layer
- Support an additional norm layer for each aux embedding. In practice, this could help stabilize the training and improve accept rate
- Gemma3/4 use nn.Embedding (Gemma3TextScaledWordEmbedding) which is not TP aware, while eagle3 drafter uses VocabParallelEmbedding. When TP>1, eagle3 drafter will get full copy of embedding.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

File: python/sglang/srt/models/llama_eagle3.py [hunk 1]
```diff
@@ -80,7 +80,6 @@ def forward(
         forward_batch: ForwardBatch,
         residual: Optional[torch.Tensor],
     ) -> Tuple[torch.Tensor, torch.Tensor]:
-
```
Comment (kpham-sgl) [inline python/sglang/srt/models/llama_eagle3.py:83]: nit

File: python/sglang/srt/models/llama_eagle3.py [hunk 2]
```diff
@@ -162,10 +180,15 @@ def forward(
...
                 assert embeds is not None
                 embeds = torch.cat(
-                    [embeds[:-1], self.embed_tokens(input_ids[-1].unsqueeze(0))]
+                    [
+                        embeds[:-1],
+                        self.embed_tokens(input_ids[-1].unsqueeze(0))
+                        * self.embed_scale,
+                    ]
                 )
+
```
Comment (kpham-sgl) [inline python/sglang/srt/models/llama_eagle3.py]: nit

File: python/sglang/srt/models/llama_eagle3.py [hunk 3]
```diff
@@ -130,11 +129,30 @@ def __init__(
             prefix=add_prefix("embed_tokens", prefix),
         )
 
+        # Embedding scale factor for target models that use scaled embeddings
+        # (e.g., Gemma3/Gemma4 multiply embeddings by hidden_size**0.5).
+        target_type = getattr(config, "target_model_type", None) or ""
+        if getattr(config, "embed_scale", None) is not None:
+            self.embed_scale = config.embed_scale
+        elif "gemma" in target_type:
+            self.embed_scale = config.hidden_size**0.5
```
Comment (kpham-sgl) [inline python/sglang/srt/models/llama_eagle3.py]: Maybe the cleaner fix is to put embed_scale in hf_transformer_utils snippet for Gemma models?

**Review by kpham-sgl — COMMENTED** [review]
One small comments and a couple of line removal/addition

File: python/sglang/srt/models/llama_eagle3.py (hunk 3 above)
Comment (pyc96) [reply python/sglang/srt/models/llama_eagle3.py]: After a second thought, I removed embed_scale part for now. Need to align SpecForge on the training side before making this change.

**Review by kpham-sgl — APPROVED** [review]

**Review by Qiaolin-Yu — APPROVED** [review]

**Comment by nvpohanh** [comment]
cc @wenscarl
